### PR TITLE
fetchEvents() to handle multiple data groups in report server

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
no issue

### Changes:
we want to fetch multiple dataGroups in report-server, ultimately we would want Aggregator to handle that.

### CODE REVIEW

File: packages/report-server/src/reportBuilder/fetch/functions/dataGroup.ts [hunk 1]
```diff
@@ -29,26 +29,32 @@ const fetchEvents = async (
...
+    }),
   );
-  const rows = response.map(event => {
-    const { dataValues, ...restOfEvent } = event;
-    return { ...dataValues, ...restOfEvent };
-  });
+
   return {
-    results: rows,
+    results: results.flat(),
```
Comment (billli0) [inline packages/report-server/src/reportBuilder/fetch/functions/dataGroup.ts:57]: I tried to use .flatMap() but it seems like TypeScript having trouble mapping the types here and throwing a type error.

File: packages/report-server/src/reportBuilder/fetch/functions/dataGroup.ts [hunk 2]
```diff
@@ -29,26 +29,32 @@ const fetchEvents = async (
...
-    aggregations,
-    organisationUnitCodes,
-    hierarchy,
-    {
-      period,
-      startDate,
-      endDate,
-    },
-    dataElementCodes,
+  // TODO: Eventually we want Aggregator to handle fetching multiple dataGroups
```
Comment (rohan-bes) [inline packages/report-server/src/reportBuilder/fetch/functions/dataGroup.ts:34]: Appreciate this comment so that we know the plan for the future! 🎉

**Review by rohan-bes — CHANGES_REQUESTED** [review]
Looks great @billli0! Can't wait to use this

One request though, I believe we also need to update the validator for `validateDataGroups()` to allow for more than 1 `dataGroup`?

**Comment by billli0** [comment]
> Looks great @billli0! Can't wait to use this
> 
> One request though, I believe we also need to update the validator for `validateDataGroups()` to allow for more than 1 `dataGroup`?

Thanks @rohan-bes , good catching!

**Review by rohan-bes — APPROVED** [review]
Awesome!

**Comment by billli0** [comment]
Note that if we fetch with `{dataGroups: ['A', 'B'], dataElements: ['A_1', 'A_2', 'B_1', 'B_2'] }`, it will send all the dataElements grouping with each dataGroup, like `{dataGroupCode: ['A'], dataElements: ['A_1', 'A_2', 'B_1', 'B_2']}`, instead of doing it with filtered dataElements like `{dataGroupCode: ['A'], dataElements: ['A_1', 'A_2']`. 

But Aggregator is smart enough to filter them itself, which is very handy. 👍

**Pull request merged**